packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx

```diff
@@ -44,17 +45,30 @@ export function DataListSort() {
   function getSortByOptions() {
     const options = sortable.reduce(
       (acc: Record<"label" | "value", string>[], sort) => {
-        const label = headers[sort];
+        const label = headers[sort.key];
         if (!label) return acc;
+        const customOptions = sort.options;
+
+        if (customOptions) {
+          customOptions.forEach(option => {
+            acc.push({
+              label: option.label || "",
+              value: `${sort.key},${option.order},${option.label}`,
+            });
+          });
 
-        if (label === "Last activity") {
-          acc.push({ label: "Last Activity (Newest)", value: `${sort},desc` });
-          acc.push({ label: "Last Activity (Oldest)", value: `${sort},asc` });
-        } else {
-          acc.push({ label: `${label} (A-Z)`, value: `${sort},asc` });
-          acc.push({ label: `${label} (Z-A)`, value: `${sort},desc` });
+          return acc;
         }
 
+        acc.push({
+          label: `${label} (A-Z)`,
+          value: `${sort.key},asc,${label}`,
+        });
+        acc.push({
+          label: `${label} (Z-A)`,
+          value: `${sort.key},desc,${label}`,
+        });
+
```

Comment:
After chatting with Chris and Darryl yesterday,  we have plans for a new way to handle these cases - where there is no dropdown custom sort but when the DataList is in mobile web, all sorting labels go to the Sort Combobox.

This will be addressed in a later PR (within this sprint)

Reply:
I don't know if I understood how it will be done, but I trust you!

but maybe before cleaning it up, we should update the implementation on JO because we are introducing breaking changes, right?

Reply:
Absolutely - I'm going to merge this and then update JO and fix the breaking changes asap